File: 8_kyu/Flick_Switch.py
import typing as typ

def flick_switch(lst :typ.Iterable[str]) -> list[bool]:
    """
    Creates list of booleans controled by strings.
    Boolean value switches if string is 'flick'.
    """
    current = True
    def flicker(s :str) -> bool:
        nonlocal current
        if s == "flick": current = not current
        return current
    return [ flicker(s) for s in lst]

File: 8_kyu/Flick_Switch.rs
/// Returns an array true/false switched by substring "flick" in the list strings.
fn flick_switch(list: &[&str]) -> Vec<bool> {
    let mut current = true;
    return
        list
        .iter()
        .map(|it| {if it.contains("flick") {current=!current}; current} )
        .collect();
}
